src/app/loadout-builder/mod-utils.ts

```diff
@@ -96,17 +96,19 @@ export function assignModsToArmorSet(
     }
   }
 
-  assignAllSeasonalMods(processItems, lockedArmor2Mods.seasonal, assignments);
-
   assignGeneralMods(
     processItems,
     lockedArmor2Mods[armor2PlugCategoryHashesByName.general],
     assignments
   );
 
-  const modsByHash = _.keyBy(Object.values(lockedArmor2Mods).flat(), (mod) => mod.mod.hash);
+  assignAllSeasonalMods(processItems, lockedArmor2Mods.seasonal, assignments);
+
+  const modsByHash = _.groupBy(Object.values(lockedArmor2Mods).flat(), (mod) => mod.mod.hash);
   const assignedMods = _.mapValues(assignments, (modHashes) =>
-    modHashes.map((modHash) => modsByHash[modHash])
```

Comment:
This was causing render key issues in components as a single mod can be assigned a number of times. Just another sneaky fix I found while I was analysing stuff.